src/main/java/com/github/fishio/PlayingField.java

```diff
@@ -199,23 +208,13 @@ public void moveMovables() {
 	 */
 	private boolean hitsWall(Entity e, ICollisionArea box) {
 		// prevent playerfish from leaving the screen
-		if (e instanceof PlayerFish) {	
-			if (box.getMaxX() >= getWidth()
-					|| box.getMinX() <= 0
-					|| box.getMaxY() >= getHeight()
-					|| box.getMinY() <= 0) {
-				return true;
-			}
+		if (e instanceof PlayerFish) {
+			return box.isOutside(0, 0, getWidth(), getHeight());
 		} else {
-			if (box.getMaxX() >= getWidth() + 2.0 * box.getWidth()
-					|| box.getMinX() <= -(2.0 * box.getWidth()) - 1
-					|| box.getMaxY() >= getHeight() + 2.0 * box.getHeight() + 1
-					|| box.getMinY() <= -(2.0 * box.getHeight()) - 1) {
-				return true;
-			}
+			double dw = 2.0 * box.getWidth() + 1;
+			double dh = 2.0 * box.getHeight() + 1;
+			return box.isOutside(-dw, -dh, getWidth() + dw, getHeight() + dh);
 		}
```

Comment:
I'm going to give you a compliment for this :+1:  That's one less responsibility in the PlayingField.

<organisation